### ISSUE TITLE
Velodyne velarray Lidar support

### ISSUE THREAD

**Issue by Magmanat** [ros-drivers/velodyne#389]
**Is your feature request related to a problem? Please describe.**
I am trying to look for a solid state lidar with a wide horizontal view but yet a decently wide vertical view that has a ros driver for obstacle avoidance for my robot
**Describe the solution you'd like**
Is anyone developing a ros driver for these new velarray sensors from velodyne velabit to m1600? 

**Describe alternatives you've considered**
Another lidar which i would use other than this might be the leddar pixel but i am not too sure if that has a ros driver either
**Additional context**

**Comment by JWhitleyWork** [comment]
The driver currently only supports spinning lidars. We would need someone with the hardware to develop the support for additional sensors.

**Comment by JKTesla** [comment]
> The driver currently only supports spinning lidars. We would need someone with the hardware to develop the support for additional sensors.

Hi buddy, do you have independently velarray Lidar ros driver(base on Ubuntu16.04 and kinetic) to share me? We are testing velarray, hope to record data using rosbag.

**Comment by jacknpainter** [comment]
Has their been any update with this issue? Are there any ROS drivers at all for the Velarray LIDARs?

**Comment by Chick92** [comment]
HAs their been any update? I have a Velarray M1600 sat on my desk if a test article is required.

**Comment by JWhitleyWork** [comment]
No pull requests have been made to support this functionality. Since this is a community-developed driver, someone with both access to the hardware and software development experience would need to add this feature and make a pull request. I am happy to review any such pull request but I don't have access to the hardware nor the time to add the feature myself.

**Comment by fjahncke** [comment]
For people looking for a M1600 driver, there are drivers available directly from Velodyne in the Vella SDK https://velodynelidar.com/products/vella/.

**Comment by ArseniiPavlov** [comment]
> For people looking for a M1600 driver, there are drivers available directly from Velodyne in the Vella SDK https://velodynelidar.com/products/vella/.

This page is not available, is there any other option to get this software?

**Comment by fjahncke** [comment]
> > For people looking for a M1600 driver, there are drivers available directly from Velodyne in the Vella SDK https://velodynelidar.com/products/vella/.
> 
> This page is not available, is there any other option to get this software?

Yes they changed the website you can now register here: https://www.vellaportal.com/register

**Comment by ArseniiPavlov** [comment]
> > > For people looking for a M1600 driver, there are drivers available directly from Velodyne in the Vella SDK https://velodynelidar.com/products/vella/.
> > 
> > 
> > This page is not available, is there any other option to get this software?
> 
> Yes they changed the website you can now register here: https://www.vellaportal.com/register

@fjahncke Thank you very much!

**Comment by mbiselx** [comment]
I wrote a small ROS driver just for the M1600 : https://github.com/mbiselx/velarray
WIth a bit of tweaking, it could also work for the 800s (I have no idea about the velabits though). 
Unfortunately my skillz are not good enough to attempt integrating this into this project (nor do I have the time right now 😢 ). If anyone wants to try porting it, they are quite welcome.
